File: php/src/Commands/Snippet.php
<?php

declare(strict_types=1);

namespace FormHandle\Commands;

use FormHandle\{Config, Output};

class Snippet
{
    public static function run(array $ctx): void
    {
        $config = Config::read();
        if ($config === null) {
            Output::error('No .formhandle config found. Run "formhandle init" first.');
            exit(1);
        }

        $resolved = Config::resolveEndpoint($config, $ctx['domain'] ?? null);
        $domain = $resolved['domain'];
        $endpoint = $resolved['endpoint'];
        $hid = $endpoint['handler_id'];

        $scriptTag = "<script src=\"https://api.formhandle.dev/s/{$hid}.js\"></script>";
        $formHtml = <<<'HTML'
<form data-formhandle>
  <input type="text" name="name" placeholder="Name" required>
  <input type="email" name="email" placeholder="Email" required>
  <textarea name="message" placeholder="Message" required></textarea>
  <button type="submit">Send</button>
</form>
HTML;

        if ($ctx['json']) {
            Output::json([
                'domain' => $domain,
                'handler_id' => $hid,
                'script_tag' => $scriptTag,
                'form_html' => $formHtml,
            ]);
        } else {
            Output::heading("Snippet for $domain");
            echo "Add this script tag to your page:\n\n";
            echo "  $scriptTag\n";
            echo "\nExample form:\n\n";
            foreach (explode("\n", $formHtml) as $line) {
                echo "  $line\n";
            }
            echo "\nAttributes:\n";
            echo '  data-formhandle-success="…"  ' . Output::dim('Custom success message') . "\n";
            echo '  data-formhandle-error="…"    ' . Output::dim('Custom error message') . "\n";
            echo "\n";
        }
    }
}

File: rust/src/commands/snippet.rs
use serde_json::json;
use std::process;

use crate::{config, output};

pub fn run(is_json: bool, domain_flag: Option<String>) {
    let cfg = config::read_config().unwrap_or_else(|| {
        output::error("No .formhandle config found. Run \"formhandle init\" first.");
        process::exit(1);
    });

    let ep = config::resolve_endpoint(&cfg, domain_flag.as_deref());
    let script_tag = format!(
        "<script src=\"https://api.formhandle.dev/s/{}.js\"></script>",
        ep.handler_id
    );
    let form_html = r#"<form data-formhandle>
  <input type="text" name="name" placeholder="Name" required>
  <input type="email" name="email" placeholder="Email" required>
  <textarea name="message" placeholder="Message" required></textarea>
  <button type="submit">Send</button>
</form>"#;

    if is_json {
        output::json_out(&json!({
            "domain": ep.domain,
            "handler_id": ep.handler_id,
            "script_tag": script_tag,
            "form_html": form_html,
        }));
    } else {
        output::heading(&format!("Snippet for {}", ep.domain));
        println!("Add this script tag to your page:\n");
        println!("  {}", script_tag);
        println!("\nExample form:\n");
        for line in form_html.lines() {
            println!("  {}", line);
        }
        println!("\nAttributes:");
        println!(
            "  data-formhandle-success=\"…\"  {}",
            output::dim("Custom success message")
        );
        println!(
            "  data-formhandle-error=\"…\"    {}",
            output::dim("Custom error message")
        );
        println!();
    }
}
